### PR TITLE
`@fiberplane/ui`: Simplify theme & provider

### PR DESCRIPTION
**Pull request by oscarvz**
# Description

- [x] No more styled-components theme values; only CSS variables
- [x] Greatly simplify `GlobalStyle` & only ship what's shared between Explorer & Studio
  - [x] Box element spacing values
  - [x] Media queries
- [x] Add variable fallback values
- [x] Documentation

# Checklist

- [x] The changes have been tested to be backwards compatible.
~~- [ ] The OpenAPI schema and generated client have been updated.~~
~~- [ ] New models module has been added to api generator xtask~~
~~- [ ] New types/fields are [well-documented and annotated](/CONTRIBUTING.md#adding-types-and-their-annotations).~~
- [x] The CHANGELOG is updated.

> Please link any related PRs in other repositories that depend on this:

### CODE REVIEW

File: fiberplane-ui/README.md [hunk 1]
```diff
@@ -7,17 +7,30 @@ Contains ReactJS UI elements for use in Fiberplane products.
...
+      type="submit"
       buttonStyle="secondary"
     >
       Click me!
     </Button>
   );
 };
 ```
+
+### Theming
```
Comment (arendjr) [inline fiberplane-ui/README.md]: It seems this heading is missing content :)

**Review by arendjr — APPROVED** [review]
Looks like a nice improvement!

Does this mean we can also get rid of the `@fiberplane/ui/dist/theme` import soon?

File: fiberplane-ui/README.md (hunk 1 above)
Comment (oscarvz) [reply fiberplane-ui/README.md]: Whoops, that was drafted but will add it in the next iteration. Will remove 🧹

**Comment by oscarvz** [comment]
@arendjr possibly, but we still want to serve an optional theme provider that we can use across our projects. The generated output from Figma doesn't contain any media query & sizing logic, but we follow the same approach across projects.
We definitely will get rid of `styled-components` as a dependency in this project however, so that'll clean up the `/theme` directory nicely. 😊

**Pull request merged**